api/tests/util/test_async_util.py

```diff
@@ -18,12 +18,14 @@
 async def queue_to_iterator(queue: asyncio.Queue[Union[T, object]]) -> AsyncIterator[T]:
     value = await queue.get()
     while value is not QUEUE_END:
-        yield value
+        # Mypy doesn't know that the only thing of type object is QUEUE_END so
+        # value is guaranteed to be of type T
+        yield value  # type: ignore
```

Comment:
We could make a type for the QUEUE_END object instead, and then we could rewrite this so mypy knows what's going on. Alternatively instead of checking if value is not QUEUE_END, you could say
`while not isinstance(value, object):`

I don't feel strongly about this, so I'll leave it up to you.

Reply:
I've done that